apps/forms-flow-ai/forms-flow-web/src/components/FOI/ReviewRequest/ReviewRequest.js

```diff
@@ -1,24 +1,179 @@
+  const [assignedToValue, setAssignedToValue] = React.useState("Unassigned");
+
+  //get the initial value of the required fields to enable/disable bottom button at the initial load of review request
+  const handleInitialRequiredRequestDescriptionValues = React.useCallback((requestDescriptionObject) => {
+    setRequiredRequestDescriptionValues(requestDescriptionObject);
+  },[])
+  const handleRequestDetailsInitialValue = React.useCallback((value) => {    
+    setRequiredRequestDetailsValues(value);
+  },[])
+  const handleAssignedToInitialValue = React.useCallback((value) => {    
+    setAssignedToValue(value);
+  },[])
+  const handleCategoryInitialValue = React.useCallback((value) => {    
+    setSelectCategoryValue(value);
+  },[])
+
+  //Update required fields of request description box with latest value
+  const handleOnChangeRequiredRequestDescriptionValues = (value, name) => {
+    const descriptionData = {...requiredRequestDescriptionValues};
+    if(name === FOI_COMPONENT_CONSTANTS.START_DATE) {      
+      descriptionData.startDate = value;      
+    }    
+    else if(name === FOI_COMPONENT_CONSTANTS.END_DATE) {    
+      descriptionData.endDate = value;      
+    }
+    else if (name === FOI_COMPONENT_CONSTANTS.DESCRIPTION) {     
+      descriptionData.description = value;      
+    }
+    else if (name === FOI_COMPONENT_CONSTANTS.IS_PROGRAM_AREA_SELECTED) {
+      descriptionData.isProgramAreaSelected = value;      
+    }
+    setRequiredRequestDescriptionValues(descriptionData);
+  }
+  
+  //Update required fields of request details box with latest value
+  const handleRequestDetailsValue = (value, name) => {
+    const detailsData = {...requiredRequestDetailsValues};
+    if (name === FOI_COMPONENT_CONSTANTS.REQUEST_TYPE) {      
+      detailsData.requestType = value;
+    }
+    else if (name === FOI_COMPONENT_CONSTANTS.RECEIVED_MODE) {
+      detailsData.receivedMode = value;
+    }
+    else if (name === FOI_COMPONENT_CONSTANTS.DELIVERY_MODE) {
+      detailsData.deliveryMode = value;
+    }    
+    setRequiredRequestDetailsValues(detailsData);
+  }
+
+  //gets the latest assigned to value
+  const handleAssignedToValue = (value) => {
+    setAssignedToValue(value);
+  }
+
+  //gets the latest category value
+  const handleCategoryValue = (value) => {
+    setSelectCategoryValue(value);
+  }
+
+  //handle email validation
+  const [validation, setValidation] = React.useState({});
+  const handleEmailValidation = (validationObj) => {    
+    setValidation(validationObj);
+  }
+
+  //to get the updated program area list with isChecked=true/false
+  const [programAreaList, setProgramAreaList] = React.useState([]);
+  
+  const handleUpdatedProgramAreaList = (programAreaList) => {    
+    //get the updated program area list with isChecked=true/false
+    setProgramAreaList(programAreaList); 
+  } 
+
+  //Variable to find if all required fields are filled or not
+  const isValidationError = (
```

Comment:
Probably smart to throw validation logic into a separate function, makes it easier to call.  Not critical to do now though;

Reply:
I will look into this